### ISSUE TITLE
Strange deformation when aligning SKEL to CMU kicking motion.

### ISSUE THREAD

**Issue by hwy1992129** [MarilynKeller/SKEL#7]
Hi, I run align SKEL with CMU 10 that having the kicking motion get the following result..
![frame_000041](https://github.com/MarilynKeller/SKEL/assets/70329303/6960e4a6-4130-46d3-b014-97b019a3ce4a)

https://github.com/MarilynKeller/SKEL/assets/70329303/4423285a-4a83-4c65-be39-c4520851f863

However, SMPL2Addbiomechanics doesn't such issue.

https://github.com/MarilynKeller/SKEL/assets/70329303/55a6d1dd-827f-4226-8cb4-620ba9e90e8e

**Comment by MarilynKeller** [comment]
Hi, glad to hear that you manage to run those two codes. Seems like the SKEL fitting does not converge on every frame, I did not have much time yet finetuning the hyperparamers. You can try the following:

- Run the fit on smaller batches, for example 20 frames (default is 3000 frames in parallel). Each batch pose will get initialized by the last frame of the previous batch result, yielding more stable results:
`python your_poses.npz -F -B 20`
This is slower, so for debugging you can add -D, which will run the fit for only the first batch and save the result. Thus you don't need to wait for the whole sequence to be processed.

-  try decreasing the learning rate and double the `max-iter` and `num_step` [here](https://github.com/MarilynKeller/SKEL/blob/2534693ebfff8964e12de478be018e2a4f704b15/skel/alignment/aligner.py#L168).

- For better debugging, you can identify the frame that did not converge in aitviewer-skel and then watch the optimization for that specific frame (here frame 215 for example, the batch size miuch be larger than this number).
`python your_poses.npz -F -w 215`

Please let me know if you find a more stable combination, I'll update the code in consequence.

**Comment by MarilynKeller** [comment]
Hi have cleaned the SKEL fit to SMPL optimization code a bit, the fit optimization should be more stable now.

Here is the result of the fit to the kick with the current version:

https://github.com/user-attachments/assets/6f92b3ea-1578-4ee6-9bc7-3da5112fd443

**Comment by hwy1992129** [comment]
Thanks for the update!
